R/estimatr_horvitz_thompson.R

```diff
@@ -283,8 +283,8 @@ horvitz_thompson <-
       # Add clusters, blocks, and treatment probabilities to data so they can be cleaned with clean_model_data
       if (!is.null(declaration$clusters)) {
         if (is.null(data[[".clusters_ddinternal"]])) {
-          data[[".clusters_ddinternal"]] <- declaration$clusters
-          clusters <- ".clusters_ddinternal"
+          .clusters_ddinternal <- declaration$clusters
+          clusters <- quo(.clusters_ddinternal)
```

Comment:
Flip the test and deindent.

Reply:
I'm not sure I follow. This makes sure we only warn in appropriate cases?

Reply:
```r
if(test){
  stop()
}
# This part not in an else block (bc stop above exits), not indented.
```

Reply:
Then we have to check a condition twice.

Reply:
```
      if (!is.null(declaration$clusters) && !is.null(data[[".clusters_ddinternal"]])) {
        stop(
          "Can't have a variable called '.clusters_ddinternal' in your `data`",
          "as we use that name for clusters from `declaration` objects."
        )
      }
      if (!is.null(declaration$clusters)) {
        .clusters_ddinternal <- declaration$clusters
        clusters <- quo(.clusters_ddinternal)
      }
```

Reply:
```
      if (!is.null(declaration$clusters) {
        if(!is.null(data[[".clusters_ddinternal"]])) {
         stop(
            "Can't have a variable called '.clusters_ddinternal' in your `data`",
            "as we use that name for clusters from `declaration` objects."
          )
        }
        .clusters_ddinternal <- declaration$clusters
        clusters <- quo(.clusters_ddinternal)
      }

Reply:
Ugh I was stuck on reworking the whole conditional sorry